File: Classes/Field.py
from typing import Dict

from .Point import Point


class Field:
    def __init__(self, data: Dict):
        self.topLeft = Point(data["topLeft"]["x"], data["topLeft"]["y"])
        self.topRight = Point(data["topRight"]["x"], data["topRight"]["y"])
        self.bottomLeft = Point(data["bottomLeft"]["x"], data["bottomLeft"]["y"])
        self.bottomRight = Point(data["bottomRight"]["x"], data["bottomRight"]["y"])

File: java/main/src/main/java/entities/Field.java
package entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Field {
    @JsonProperty("bottomLeft")
    private Point bottomLeft;
    @JsonProperty("bottomRight")
    private Point bottomRight;
    @JsonProperty("topLeft")
    private Point topLeft;
    @JsonProperty("topRight")
    private Point topRight;

    public Field(Point bottomLeft, Point bottomRight, Point topLeft, Point topRight) {
        this.bottomLeft = bottomLeft;
        this.bottomRight = bottomRight;
        this.topLeft = topLeft;
        this.topRight = topRight;
    }

    public Point getBottomLeft() {
        return bottomLeft;
    }

    public void setBottomLeft(Point bottomLeft) {
        this.bottomLeft = bottomLeft;
    }

    public Point getBottomRight() {
        return bottomRight;
    }

    public void setBottomRight(Point bottomRight) {
        this.bottomRight = bottomRight;
    }

    public Point getTopLeft() {
        return topLeft;
    }

    public void setTopLeft(Point topLeft) {
        this.topLeft = topLeft;
    }

    public Point getTopRight() {
        return topRight;
    }

    public void setTopRight(Point topRight) {
        this.topRight = topRight;
    }
}
